sql/sql_utils.cc

```diff
@@ -38,6 +43,64 @@ std::string escapeSql(const char * str)
     return escapeSql(std::string(str));
 }
 
+bool matchSqlFilter(const Utf8String& valueString, const Utf8String& filterString)
+{
+    if (filterString.empty() || valueString.empty())
+        return false;
+
+    Utf8String regExFilter;
+
+    for (const auto& filterChar : filterString) {
+
+        switch (filterChar) {
+            case ('%'): {
+                regExFilter += ".*";
+                break;
+            }
+            case ('_'): {
+                regExFilter += ".";
+                break;
+            }
+            case ('*'): {
+                regExFilter += "[*]";
+                break;
+            }
+            case ('['): {
+                regExFilter += "\\[";
+                break;
+            }
+            case (']'): {
+                regExFilter += "\\]";
+                break;
+            }
+            case ('.'): {
+                regExFilter += "[.]";
+                break;
+            }
+            case ('|'): {
+                regExFilter += "[|]";
+                break;
+            }
+            case ('('): {
+                regExFilter += "[()]";
```

Comment:
i don't know what this case statement does but shouldn't this be `[(]` instead of the current `[()]` ?